newrelic/resource_newrelic_application_test.go

```diff
@@ -0,0 +1,71 @@
+package newrelic
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccNewRelicApplication_Basic(t *testing.T) {
+	resourceName := "newrelic_application.app"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckNewRelicApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNewRelicApplicationConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNewRelicApplicationExists(testAccExpectedApplicationName),
+					resource.TestCheckResourceAttr(
+						resourceName, "name", testAccExpectedApplicationName),
+					resource.TestCheckResourceAttr(
+						resourceName, "app_apdex_threshold", "0.9"),
+					resource.TestCheckResourceAttr(
+						resourceName, "end_user_apdex_threshold", "0.8"),
+					resource.TestCheckResourceAttr(
+						resourceName, "enable_real_user_monitoring", "true"),
+				),
+			},
+			// Test: Import
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckNewRelicApplicationDestroy(s *terraform.State) error {
+	// We expect the application to still exist
+	return nil
+}
+
+// The test application for this data source is created in provider_test.go
+func testAccNewRelicApplicationConfig() string {
+	return fmt.Sprintf(`
+resource "newrelic_application" "app" {
+	name = "%s"
+	app_apdex_threshold = "0.9"
+	end_user_apdex_threshold = "0.8"
+	enable_real_user_monitoring = true
+}
+`, testAccExpectedApplicationName)
+}
+
+func testAccCheckNewRelicApplicationExists(n string) resource.TestCheckFunc {
```

Comment:
I think we want to make a call to the API via the client to ensure it exists. At least that's what we're doing in the other resources I believe. Should be relatively easy copy and paste and replace a few things.

Reply:
Have done.